### DISCUSSION TITLE
My setup (HM-1500, HM-800, HM-600, US5000, R4850G, 3xOpenDTU)

### DISCUSSION THREAD

**Discussion by bf4ed** [hoylabs/OpenDTU-OnBattery#515] [Show and tell]
On April 2023 i purchased HM-600 with 2 Panels. After search for OpenDTU i stumbled on this project, with Battery.
I then purchased HM-1500 and two additional Panels, the spare HM-600 was intended for Battery.
After consideration i purchased another Hoymiles, the HM-800. And here is my setup after completion in August 2023 and testing:

- 2x ESP32 (for OpenDTU-Night, OpenDTU-AUX) with nRF24+, 1,3” OLED Display, SN65HVD230 and MCP2515
- 1x ESP32 (for OpenDTU-Day) with nRF24+
- 1x Hoymiles HM-1500 (Day)
- 1x Hoymiles HM-800 (Night)
- 1x Hoymiles HM-600 (AUX)
- 1x Pylontech US5000 (48V, 4,8kWh)
- 1x Shelly 3EM (Grid consumption IN/OUT)
- 1x Shelly EM (Solar production HM-1500)
- 2x Shelly 1PM (HM-800, HM-600 for consumption and remote ON/OFF)
- 1x Shelly 2PM (Huawei R4850G for consumption and remote ON/OFF(CH1), soft OFF (CH2))

On the House is the HM-1500 with OpenDTU-Day. This one is only for solar production and is set to -600W. I figured out that it is better to have the inverter to run with -600W than with 0W to Grid export. It is more efficient on sunrise and sunset.
The other setup is in the Garage.
HM-800 with OpenDTU-Night is for night consumption, to regulate charging with Huawei R4850G over CAN Bus and monitoring Pylontech Battery over CAN Bus.
HM-600 with OpenDTU-AUX is for auxiliary Power when needed.
I also can select in which phase i connect the inverter when needed (the 1-2-3 Switch on left). 

Thanks to the makers of OpenDTU and all others in this community!!! Really great project.

![00_bf4ed-Solar_Setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/b7730a93-33f4-4342-8adc-bcd9db17a130)

Testing the setup
![01_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/06b617a3-c90b-450e-8075-b6d8d2fb28f0)

Building the housing for HM-800
![02_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/5f90b4b7-601b-4470-bf30-8489d45fb23a)
![03_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/97f937e5-735b-44ad-a104-711e6ab13f8d)
![04_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/dbee9b99-bb39-4ba1-b1cc-92ea0e5071e0)
![05_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/db58b9e0-b8e5-4caa-b32a-8b97623aefb2)
![06_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/f4194d08-111f-42d7-b044-cfc485a456b6)
![07_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/3e8ff054-4dfc-4cfa-9443-c640bed5345c)
![08_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/b3c8f1f5-3e49-4b66-a59e-b595b27f47b5)
![09_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/74ff6332-50f2-43b8-bc4f-421c5b72610d)

Housing for Huawei charger (Pull out hot swapable like the original :) )
![10_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/3224c13b-73db-44c8-b68c-e4c9afe33b78)
![11_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/b1f6ee2e-d93e-491e-925f-21548e5b62b9)

Housing for HM-600
![12_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/21ede34e-d46e-4ef0-905a-be964453e467)
![13_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/334b949a-9cdb-432d-b6b7-8c90daa5f7be)

The OpenDTU board, OLED is detachable for testing, later with screw connectors on the front
![14_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/9e79f1fa-4716-4bcf-8ce9-44e033e78fb9)
![15_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/04b647e4-ad5a-42c5-931b-ac0c6a6c5a4b)
![16_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/cc601cba-2195-4c9f-ba64-b45cdbb653ca)
![17_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/7b691e9f-66b7-47fc-938d-818138745a62)

The 19" cabinet with 3F 13A automatic fuse, thermostat switch is for the power supply for fans on top
![18_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/6b76eba6-51d2-465c-910e-c1084b179b82)
![19_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/e268e191-3653-43fd-a03e-4418a8140ab3)
![20_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/4d1a714a-ed8a-45de-bbb3-ed41eb9a5418)

Complete, AC IN is 3F selectable on inverters with the switch on the left
![21_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/cbc1bfb1-9d9f-4ea0-8e30-80c3d3256acf)

And this is the OpenDTU-Day for the HM-1500 in the house
![22_bf4ed-solar_setup](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/73cbf38d-ab00-43b2-a443-26d6054fc8bf)

**Comment by Manos1966** [comment]
**R E S P E K T !**

**Comment by Manos1966** [comment]
Hallo @bf4ed 
Can you post some explanations regarding the settings of the OpenDTU that controls the R4850G ?
Especially the settings on the Power Limiter Page.
German or English is OK, also screenshots are enough.
(I want to add some information to the Wiki).

RGDS
Manos

**Comment by bf4ed** [comment]
![Screenshot_20231207-025938_cropped](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/64aa21de-753a-437a-9a60-269133f10151)
![Screenshot_20231207-025954_cropped](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/94ad097f-09c7-4267-a9d8-3402d2941793)
![Screenshot_20231207-030038_cropped](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/bcd724e5-ae6f-4293-9235-415556d8ea28)
![Screenshot_20231207-030109_cropped](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/4ad7b624-b362-4eff-a924-dfa2528ae626)
![Screenshot_20231207-030218_cropped](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/130218342/fc1b5458-e71c-4928-805c-619101f9b9e7)

This is OpenDTU-Night
Power meter is a Shelly 3EM

**Comment by Manos1966** [comment]
Many thanks for screenshots @bf4ed 

I must admit, I look at the pictures of your constructions and I get depressed 🙈 it is so far away from any possibility I may ever have to construct something like this myself.... 😭 

Just to make sure I understand the mode of function correctly:
1. HM-1500 with  **OpenDTU-Day** during the day produces and is set to -600W. 

2. During the day, the HM-800 with **OpenDTU-Night** is regulating the Huawei R4850 and charges the Pylontech.
So, during the day if the Shelly 3em says -600W, the openDTU-Night will order the R4850 to start charging with 600W, correct?
The Shelly 3em says 0W thus the OpenDTU-Day will order the HM1500 to increase production to reach -600W again.
The Shelly 3em will say -600W again, at which moment the OpenDTU-Night will order the R4850 to start charging with 600+600=1200W
The Shelly 3em says 0W again thus the OpenDTU-Day will order the HM1500 to increase production to reach -600W again which is not possible because it is a HM1500 thus it increases another 300W (600W+600W+300W = 1500).
The Shelly 3em will say -300W at which moment the OpenDTU-Night will order the R4850 to start charging with 600+600+300=1500W
(I did not mention the power consumption of the household nor that your East/West configuration will always deliver much less than 1500W at any time, for simplification reasons).
Genius!

3. What I did not understand yet is: How do you start the HM-800 only at night?
With the Shelly plus 1PM on the Hutschiene I guess? 
What kind of settings do you have on that Shelly? Fixed time?

**Reply by bf4ed** [reply]
1. Correct (From the beginning i have it at -30W, but then the MPPT is not working correctly, i get much less power, when the sun is weak. Therefore -600W.)
2. Correct
3. Correct (The Shelly1PM is for controlling the output. I have a 10W/230V fan for cooling on output from inverter and to not waste energy i turn it OFF. The timer is set from ON 19-07h in Sommer and now in Winter ON 16-07h.)

**Reply by Manos1966** [reply]
Many Thanks!

**Comment by dragricola** [comment]
Hello @bf4ed and @Manos1966 
Thank you very much! 
The settings you published helped me to put my very similar system to operation. 

One small detail can be added in your documentation: For systems that communicate via MQTT with a home automation system it is important to set the MQTT Topics powerlimiter/cmd/mode to 0, huawei.cmd.mode to 3 and the other Topics under huawei.cmd.limit_... to the battery depending values.
